bugbot/rules/bugscore.py

```diff
@@ -0,0 +1,144 @@
+
+    def columns(self):
+        return [
+            "id",
+            "summary",
+            "creation",
+            "last_change",
+            "severity",
+            "dups_count",
+            "votes",
+            "cc_count",
+            "see_also_count",
+            "user_impact_score",
+            "comment_count",
+            "product",
+            "component",
+            "type",
+        ]
+
+    def get_extra_for_template(self):
+        return {
+            "dups_threshold": self.ndups,
+            "votes_threshold": self.votes,
+            "cc_threshold": self.cc,
+            "see_also_threshold": self.see_also,
+            "comments_threshold": self.comments,
+            "votes_weight": self.votes_weight,
+            "see_also_weight": self.see_also_weight,
+            "dups_weight": self.dups_weight,
+            "comments_weight": self.comments_weight,
+            "ccs_weight": self.ccs_weight,
+            "threshold": self.threshold,
+        }
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        cc_count = len(bug["cc"])
+        dups_count = len(bug["duplicates"])
+        votes_count = bug["votes"]
+        see_also_count = len(bug["see_also"])
+        comment_count = bug["comment_count"]
+
+        # Calculating the User Impact Score
+        user_impact_score = round(
+            (votes_count * self.votes_weight)
+            + (see_also_count * self.see_also_weight)
+            + (dups_count * self.dups_weight)
+            + (comment_count * self.comments_weight)
+            + (cc_count * self.ccs_weight)
+        )
+        if user_impact_score < self.threshold:
+            return
+        data[bugid] = {
+            "creation": utils.get_human_lag(bug["creation_time"]),
+            "last_change": utils.get_human_lag(bug["last_change_time"]),
+            "severity": bug["severity"],
+            "dups_count": dups_count,
+            "votes": votes_count,
+            "cc_count": cc_count,
+            "see_also_count": see_also_count,
+            "user_impact_score": user_impact_score,
+            "comment_count": comment_count,
+            "product": bug["product"],
+            "component": bug["component"],
+            "type": bug["type"],
+        }
+
+        return bug
+
+    def get_bz_params(self, date):
+        fields = [
+            "creation_time",
+            "last_change_time",
+            "severity",
+            "votes",
+            "cc",
+            "duplicates",
+            "see_also",
+            "comment_count",
+            "product",
+            "component",
+            "type",
+        ]
+
+        params = {
+            "include_fields": fields,
+            "resolution": "---",
+            "f1": "keywords",
+            "o1": "nowords",
+            "v1": ["meta", "intermittent"],
+            "j3": "OR",
+            "f3": "OP",
+            "f4": "dupe_count",
+            "o4": "greaterthaneq",
+            "v4": "5",
```

Comment:
nit: Declaring these thresholds in the constructor after the highlighting thresholds will help distinguish between the two types of thresholds.